jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/RunConfigTestUtils.kt

```diff
@@ -48,19 +46,12 @@ fun executeRunConfiguration(runConfiguration: RunConfiguration, executorId: Stri
         val executionEnvironment = ExecutionEnvironmentBuilder.create(executor, runConfiguration).build {
             it.processHandler?.addProcessListener(
                 object : OutputListener() {
-                    override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
-                        // Ansi codes throw off the default logic, so remap it to check the base type
-                        val processOutputType = outputType as? ProcessOutputType
-                        val baseType = processOutputType?.baseOutputType ?: outputType
-
-                        super.onTextAvailable(event, baseType)
-
-                        println("[${if (baseType == ProcessOutputTypes.STDOUT) "stdout" else "stderr"}]: ${event.text}")
-                    }
-
                     override fun processTerminated(event: ProcessEvent) {
                         super.processTerminated(event)
-                        executionFuture.complete(this.output)
+                        // if using the default step executor as the process handle, need to pull text from it or it will be empty
+                        // otherwise, pull the output from the process itself
+                        val output = (it.processHandler as? StepExecutor.StepExecutorProcessHandler)?.getFinalOutput() ?: this.output
```

Comment:
Interesting...this (`it.processHandler`) should be able to have a CapturingProcessAdaper added as a listener since the base class of the dummy (now StepExecutorProcessHandler) still uses the event multicaster. This would remove you ahving to do the finalOutput stuff yourself.

Reply:
It would be to keep the production code cleaner / not waste the memory since the text output is only for tests but if its not possible, then its not possible